flutter_mobx_forms/lib/src/form_field.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:mobx/mobx.dart';
+
+part 'form_field.g.dart';
+
+typedef SyncFieldValidator<T> = Iterable<String> Function(T value);
+typedef AsyncFieldValidator<T> = Stream<String> Function(T value);
+
+enum ValidationPolicy { onChange, always, manual }
+
+class FormField<T> = _FormField<T> with _$FormField;
+
+abstract class _FormField<T> with Store {
+  _FormField(
+      {this.name, this.label, this.value, this.validator, this.asyncValidator});
+
+  final String name;
+
+  final String label;
+
+  @observable
+  List<String> errors;
+
+  @observable
+  T value;
+
+  @observable
+  bool _isValidating;
+
+  @computed
+  bool get isValidating => _isValidating;
+
+  @computed
+  String get error => errors == null ? null : errors[0];
```

Comment:
i know it's not ready for review but should the condition here should be `errors == null || errors.isEmpty` so it's more safe?

Reply:
No, this is about extracting the first `error` in the list of `errors`, but yes we need to do the `isEmpty` check as well to ensure we don't overflow